app/src/main/java/com/lagradost/cloudstream3/extractors/Rabbitstream.kt

```diff
@@ -0,0 +1,167 @@
+            referer = mainUrl,
+            headers = mapOf("X-Requested-With" to "XMLHttpRequest")
+        )
+        val encryptedMap = response.parsedSafe<SourcesEncrypted>()
+        val sources = encryptedMap?.sources
+        val decryptedSources = if (sources == null || encryptedMap.encrypted == false) {
+            response.parsedSafe()
+        } else {
+            val (key, encData) = extractRealKey(sources, rawKey)
+            val decrypted = decryptMapped<List<Sources>>(encData, key)
+            SourcesResponses(
+                sources = decrypted,
+                tracks = encryptedMap.tracks
+            )
+        }
+
+        decryptedSources?.sources?.map { source ->
+            M3u8Helper.generateM3u8(
+                name,
+                source?.file ?: return@map,
+                "$mainUrl/",
+            ).forEach(callback)
+        }
+
+        decryptedSources?.tracks?.map { track ->
+            subtitleCallback.invoke(
+                SubtitleFile(
+                    track?.label ?: "",
+                    track?.file ?: return@map
+                )
+            )
+        }
+
+    }
+
+    private fun extractRealKey(originalString: String?, stops: String): Pair<String, String> {
+        val table = parseJson<List<List<Int>>>(stops)
+        val decryptedKey = StringBuilder()
+        var offset = 0
+        var encryptedString = originalString
+
+        table.forEach { (start, end) ->
+            decryptedKey.append(encryptedString?.substring(start - offset, end - offset))
+            encryptedString = encryptedString?.substring(
+                0,
+                start - offset
+            ) + encryptedString?.substring(end - offset)
+            offset += end - start
+        }
+        return decryptedKey.toString() to encryptedString.toString()
+    }
+
+    private inline fun <reified T> decryptMapped(input: String, key: String): T? {
+        val decrypt = decrypt(input, key)
+        return AppUtils.tryParseJson(decrypt)
+    }
+
+    private fun decrypt(input: String, key: String): String {
+        return decryptSourceUrl(
+            generateKey(
+                base64DecodeArray(input).copyOfRange(8, 16),
+                key.toByteArray()
+            ), input
+        )
+    }
+
+    private fun generateKey(salt: ByteArray, secret: ByteArray): ByteArray {
+        var key = md5(secret + salt)
+        var currentKey = key
+        while (currentKey.size < 48) {
+            key = md5(key + secret + salt)
+            currentKey += key
+        }
+        return currentKey
+    }
+
+    private fun md5(input: ByteArray): ByteArray {
+        return MessageDigest.getInstance("MD5").digest(input)
+    }
+
+    private fun decryptSourceUrl(decryptionKey: ByteArray, sourceUrl: String): String {
+        val cipherData = base64DecodeArray(sourceUrl)
+        val encrypted = cipherData.copyOfRange(16, cipherData.size)
+        val aesCBC = Cipher.getInstance("AES/CBC/PKCS5Padding")
```

Comment:
add `?: throw ErrorLoadingExeption("Cipher not found")`

so you dont need to do Objects.requireNonNull and !! later